src/components/PartRow.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  Checkbox,
+  Collapse,
+  IconButton,
+  List,
+  ListItem,
+  ListItemText,
+  Table,
+  TableBody,
+  TableContainer,
+  TableHead,
+  TableCell,
+  TableRow,
+  Typography,
+} from "@material-ui/core";
+import { KeyboardArrowDown, KeyboardArrowUp } from "@material-ui/icons";
+import React from "react";
+import { Part } from "../lib/parts";
+import { PartRevision } from "../lib/part-revisions";
+import { toPartRevisionData as toPartRevisionPage } from "../lib/part-revisions";
+import { GetPartRevisionsRes } from "../pages/api/part-revisions";
+import { Paged } from "../lib/paging";
+
+interface PartRowProps {
+  part: Part;
+  isSelected: boolean;
+  onSelected: (id: string) => void;
+}
+
+async function fetcher(req: RequestInfo) {
+  return (await fetch(req)).json();
+}
+
+export default function PartRow({
+  part,
+  isSelected,
+  onSelected,
+}: PartRowProps): JSX.Element {
+  const row = part;
+  const [open, setOpen] = React.useState<boolean>(false);
+
+  const [revisions, setRevisions] = React.useState<
+    Paged<PartRevision> | undefined
+  >();
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      const result = await fetch(`/api/part-revisions?partId=${part.id}`);
+
+      const j = (await result.json()) as GetPartRevisionsRes;
+
+      setRevisions(toPartRevisionPage(j));
+    };
+
+    if (open) {
+      fetchData();
+    }
+  }, [part.id, open]);
+
+  return (
+    <>
+      <TableRow
+        hover
+        role="checkbox"
+        tabIndex={-1}
+        key={row.id}
+        selected={isSelected}
+        sx={{ "& > *": { borderBottom: "unset" } }}
+      >
+        <TableCell>
+          <IconButton
+            aria-label="expand row"
+            size="small"
+            onClick={() => setOpen(!open)}
+          >
+            {open ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
+          </IconButton>
+        </TableCell>
+        <TableCell padding="checkbox" onClick={() => onSelected(row.id)}>
+          <Checkbox color="primary" checked={isSelected} />
+        </TableCell>
+        <TableCell>{row.name}</TableCell>
+        <TableCell>{row.suppliedId}</TableCell>
+        <TableCell>{row.id}</TableCell>
+        <TableCell>
+          {row.created ? new Date(row.created).toLocaleString() : undefined}
+        </TableCell>
+      </TableRow>
+      <TableRow>
+        <TableCell
+          style={{ paddingBottom: 0, paddingTop: 0 }}
```

Comment:
Nit, you can use pb and pt, pr and pl. Same with margins.

Reply:
Updated